src/modules/auth/validation/authValidations.ts

```diff
@@ -0,0 +1,76 @@
+import Joi from "joi";
+
+interface User {
+    firstName: string;
+    lastName: string;
+    email: string;
+    password: string;
+    phone: number;
+    profilePicture?: string;
+    gender: "male" | "female" | "other";
+    birthDate: string;
+    language: string;
+    currency: string;
+    role: "buyer" | "seller" | "admin";
+}
+
+const authSchema = Joi.object<User>({
+    firstName: Joi.string().required().messages({
+        "string.base": "First name should be a type of text",

```

Comment:
make sure to return the exact attribute name into the message. 
i.e, If attribute name is ```firstName```. The message should be "firstName is required"

```
firstName: Joi.string().required().messages({
        "string.base": "firstName should be a type of text",
        "string.empty": "firstName cannot be an empty field",
        "any.required": "firstName is required"
    }),
```